docs/tutorial/connecting_widgets_to_a_function.md

```diff
@@ -0,0 +1,184 @@
+To make a widget interactive, you need to connect it to a function. We can do this by using a signal-slot mechanism. A signal is emitted when a particular event occurs, and a slot is a function that is called in response to the signal. In PySide6, you can connect a signal to a slot using the `connect` method of the signal object. 
+
+<hr>
+
+
+## Creating a slot function
+
+The slot function doesn't take any arguments. To handle this, we can group all the widgets together with all slot functions in a single class. This way, we can access the widgets inside the slot function through the `self` reference. Assuming that the following code is inside a class in which a [`QLabel`](../../QtWidgets/Qlabel) widget is defined as `self.label`, we can define a slot function as follows:
+
+
+```python
+@Slot()
+def change_label_text(self):
+    self.label.setText("Hello, PySide6!")
+```
+
+In the code above, we use the `@Slot()` decorator to indicate that the function is a slot. So, if we want to connect this slot function to the signal of the `QPushButton` in our `simple_gui.py` file, we first need to change up our code quite a bit. We can put all the code in a class that inherits from [`QMainWindow`](../../QtWidgets/QMainWindow) and let's call it `UserInterface`: 
+
+=== "Without class"
+
+    ```python title="simple_gui.py" linenums="1"
+    import PySide6.QtWidgets as QtWidgets
+    from PySide6.QtCore import Slot
+
+
+    app = QtWidgets.QApplication()
+    window = QtWidgets.QMainWindow()
+
+    # Setting the central widget 
+    central_widget = QtWidgets.QWidget()
+    window.setCentralWidget(central_widget)
+
+    # Adding a layout to the central widget
+    vbox = QtWidgets.QVBoxLayout()
+    central_widget.setLayout(vbox)
+
+    # Creating a button and a label
+    label = QtWidgets.QLabel()
+    label.setText("Hello, User! Please click the button.")
+    button = QtWidgets.QPushButton()
+    button.setText("Click me!")
+
+    # Adding the button and label to the layout
+    vbox.addWidget(label)
+    vbox.addWidget(button)
+
+    window.show()
+    app.exec()
+    ```
+
+===+ "With class"
```

Comment:
The step towards a whole class is really big for only adding functionallity to the button.

We could add the functionallity to the without class like this:

```
@Slot()
def change_label_text():
        label.setText("Hello, PySide6!")

# Connecting the button to the slot function
button.clicked.connect(change_label_text)
```

Maybe it is not neat to do it without a class or are there other problems? (@davidfokkema why do you want to use a class when constructing a GUI?)

Then we could add a 4th step where we go from the plain Python code to the class? In that way step 3 is a bit smaller.

Reply:
Yes, the no-class code would be:
```python
from PySide6 import QtWidgets
from PySide6.QtCore import Slot


app = QtWidgets.QApplication()
window = QtWidgets.QMainWindow()

# Setting the central widget
central_widget = QtWidgets.QWidget()
window.setCentralWidget(central_widget)

# Adding a layout to the central widget
vbox = QtWidgets.QVBoxLayout()
central_widget.setLayout(vbox)

# Creating a button and a label
label = QtWidgets.QLabel()
label.setText("Hello, User! Please click the button.")
button = QtWidgets.QPushButton()
button.setText("Click me!")

# Adding the button and label to the layout
vbox.addWidget(label)
vbox.addWidget(button)

@Slot()
def change_label_text():
        label.setText("Hello, PySide6!")

# Connecting the button to the slot function
button.clicked.connect(change_label_text)

window.show()
app.exec()
```

Reply:
So... _why_ use classes? They help to organise your code. They bundle data, child objects and methods in a single container. You can have a MainWindow, FileDialog, Plot, etc. each with their own child widgets, some variables and behaviour (slots and signals) all wrapped up in a package. Crucially, they can _all_ have their own `button`. Without classes, there can only be _one_ `button` since everything is in the global namespace. You can also have more of them: two FileDialogs, three Plots, etc. In the example above you're mixing imports, initialisation code, a function definition and the code that starts your application (app.exec) in the same file without any structure whatsoever.

Reply:
I second Annelies' suggestion: first introduce slots and signals, then comment on the fact that the code is already becoming quite messy and point out some potential sources of error and frustration (the single `button`, you cannot have any more `button`s; each one of them has to be uniquely named throughout your entire application) and then rework the code into a class.